adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m

```diff
@@ -0,0 +1,257 @@
+
+  /// The Vungle native ad.
+  VungleNativeAd *_nativeAd;
+
+  /// The ad event delegate to forward ad rendering events to the Google Mobile Ads SDK.
+  id<GADMediationNativeAdEventDelegate> _delegate;
+    
+  /// The Vungle container for the main ad
+  VungleMediaView *_mediaView;
+}
+
+@synthesize desiredPlacement;
+@synthesize isAdLoaded;
+
+- (nonnull instancetype)initNativeAdForAdConfiguration:(nonnull GADMediationNativeAdConfiguration *)adConfiguration
+                                     completionHandler:(nonnull GADMediationNativeLoadCompletionHandler)completionHandler {
+  self = [super init];
+  if (self) {
+    _adConfiguration = adConfiguration;
+
+    // Store the ad config and completion handler for later use.
+    __block atomic_flag adLoadHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationNativeLoadCompletionHandler originalCompletionHandler = [completionHandler copy];
+
+    // Ensure the original completion handler is only called once, and is deallocated once called.
+    _adLoadCompletionHandler = ^id<GADMediationNativeAdEventDelegate>(_Nullable id<GADMediationNativeAd> ad, NSError *_Nullable error) {
+      if (atomic_flag_test_and_set(&adLoadHandlerCalled)) {
+        return nil;
+      }
+      id<GADMediationNativeAdEventDelegate> delegate = nil;
+      if (originalCompletionHandler) {
+        delegate = originalCompletionHandler(ad, error);
+      }
+      originalCompletionHandler = nil;
+      return delegate;
+    };
+  }
+  return self;
+}
+
+- (void)requestNativeAd {
+  self.desiredPlacement = [GADMAdapterVungleUtils findPlacement:_adConfiguration.credentials.settings
+                                                  networkExtras:_adConfiguration.extras];
+  if (!self.desiredPlacement) {
+    NSError *error = GADMAdapterVungleErrorWithCodeAndDescription(GADMAdapterVungleErrorInvalidServerParameters, @"Placement ID not specified.");
+    _adLoadCompletionHandler(nil, error);
+    return;
+  }
+    
+  if (![[VungleSDK sharedSDK] isInitialized]) {
+    NSString *appID = [GADMAdapterVungleUtils findAppID:_adConfiguration.credentials.settings];
+    [[GADMAdapterVungleRouter sharedInstance] initWithAppId:appID delegate:self];
+    return;
+  }
+    
+  [self loadAd];
+}
+
+- (void)loadAd {
+  _nativeAd = [[VungleNativeAd alloc] initWithPlacementID:self.desiredPlacement];
+  _nativeAd.delegate = self;
+  VungleAdNetworkExtras *networkExtras = [_adConfiguration extras];
+  if (networkExtras.nativeAdOptionPosition == 1) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionTopLeft;
+  } else if (networkExtras.nativeAdOptionPosition == 2) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionTopRight;
+  } else if (networkExtras.nativeAdOptionPosition == 3) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionBottomLeft;
+  } else if (networkExtras.nativeAdOptionPosition == 4) {
+    _nativeAd.adOptionsPosition = NativeAdOptionsPositionBottomRight;
+  }
```

Comment:
Nit, perhaps we can have a `switch` statement here.

In addition, what should the value of `_nativeAd.adOptionsPosition` be if the publishers don't set this extra?

Reply:
The position will be the default value if not set, which is the top right.

Reply:
Thanks Stanley.

Looking at [Vungle documentation](https://support.vungle.com/hc/en-us/articles/4416477036955#configure-the-native-ad-for-display-using-storyboard-0-6), it says:
```
Before calling the method to display the Native Ad, you must add a line of code to specify the 
position of the privacy icon. This position should be one of the available positions in the 
NativeAdOptionsPosition enum found in VungleNativeAd.h.
```

Is that not the case here?

Reply:
Gentle bump @stanley-vungle @ManojBudumuru-vungle.

Reply:
Another bump.

Reply:
@joshh-devrel Updated. please check. thanks